### PR TITLE
Feature/weapon refinements

### PR DESCRIPTION
**Pull request by WillTarte**
Added set of official weapons:
- Assault Riffle
- Sniper
- Shotgun
- Knife
- Grenade

Values can be tweaked ofc, but It'll be easier to balance when the game is almost complete and animations are done

### CODE REVIEW

**Review by ClaudiaLapalme — CHANGES_REQUESTED** [review]
1. I see that the knife mechanics have not been implemented yet. Is this going to be in a separate PR?

2. I see that throwables don't destroy the environment; it might be a good idea to add that since it's already in a todo

3. please no massive refactoring in a non-refactoring PR next time

**Comment by WillTarte** [comment]
> 1. I see that the knife mechanics have not been implemented yet. Is this going to be in a separate PR?
> 2. I see that throwables don't destroy the environment; it might be a good idea to add that since it's already in a todo
> 3. please no massive refactoring in a non-refactoring PR next time

1. The knife melee weapon is implemented.
2. That's gonna be implemented by issue #13.
3. The point of this PR is to have a MVP weapons system.

File: Assets/Scripts/MonoBehaviours/WeaponsSystem/ProjectileBehaviourScript.cs [hunk 1]
```diff
@@ -38,19 +44,28 @@ private void Awake()
...
         {
-            if (_projectileData.IsKinematic)
+            transform.Translate(_projectileData.ProjectileSpeed * Time.deltaTime * _direction);
+        }
+
+        private void OnTriggerEnter2D(Collider2D other)
+        {
+            if (other.CompareTag("Enemy"))
+            {
+                //todo: Deal damage to enemy
```
[the remark this answers is not in the document]
Comment (WillTarte) [reply Assets/Scripts/MonoBehaviours/WeaponsSystem/ProjectileBehaviourScript.cs:59]: Yessir

File: Assets/Scripts/MonoBehaviours/WeaponsSystem/ThrowableBehaviourScript.cs [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                if (hit.CompareTag("Enemy"))
+                {
+                    // todo: do damage to enemy
+                }
+                else if (hit.CompareTag("Player"))
+                {
+                    var playerController = hit.GetComponent<Controller>();
+                    playerController.LoseHitPoints(_projectileData.ProjectileBaseDamage);
+                }
+                // todo: interacts with breakable things
```
[the remark this answers is not in the document]
Comment (WillTarte) [reply Assets/Scripts/MonoBehaviours/WeaponsSystem/ThrowableBehaviourScript.cs:88]: That's a different issue #13

File: Assets/Scripts/MonoBehaviours/WeaponsSystem/ProjectileBehaviourScript.cs [hunk 3]
```diff
@@ -7,6 +9,7 @@ namespace MonoBehaviours.WeaponsSystem
     /// <summary>
     /// Controls the behavior of a projectile
     /// </summary>
+    // todo: Environment needs to be on different layers, and projectiles only interact with some of the environment layers
```
[the remark this answers is not in the document]
Comment (WillTarte) [reply Assets/Scripts/MonoBehaviours/WeaponsSystem/ProjectileBehaviourScript.cs:12]: It's gonna be done through #13

File: Assets/Scripts/MonoBehaviours/WeaponsSystem/ProjectileBehaviourScript.cs [hunk 4]
```diff
@@ -38,19 +44,28 @@ private void Awake()
...
+            }
+            else if (_shouldDamagePlayer && other.CompareTag("Player"))
+            {
+                var player = other.GetComponent<Controller>();
+                player.LoseHitPoints(_projectileData.ProjectileBaseDamage);
+            } 
+            else if (other.CompareTag("Ground") || other.CompareTag("Platform"))
             {
-                transform.Translate(_projectileData.ProjectileSpeed * Time.deltaTime * _direction);
+                Destroy(gameObject);
```
[the remark this answers is not in the document]
Comment (WillTarte) [reply Assets/Scripts/MonoBehaviours/WeaponsSystem/ProjectileBehaviourScript.cs:68]: This destroys the projectile gameobject, not the environment

**Comment by ClaudiaLapalme** [comment]
The shotgun bullet speed is insanely slow. Would it be possible to just fine-tune the speed of the ammunition?

**Review by ClaudiaLapalme — APPROVED** [review]

**Pull request merged**